src/patterns/security-code/index.md.njk

```diff
@@ -0,0 +1,135 @@
+---
+title: Confirm a phone number
+description: Identifying users when they sign in
+section: Patterns
+theme: Help users to…
+aliases: 2FA, MFA, multi-factor authentication, security code, text message, two-factor authentication
+backlog_issue_id: 25
+layout: layout-pane.njk
+---
+
+{% from "_example.njk" import example %}
+
+Check that a user has access to a specific mobile phone number using a security code sent by text message.
+
+{{ example({group: "patterns", item: "security-code", example: "default", html: true, nunjucks: true, open: false}) }}
+
+## When to use this pattern
+
+Ask users to enter a security code when they need to sign in or complete a higher-risk task, such as changing a password.
+
+Asking users to enter a security code sent to their mobile phone gives a second layer of security over a [password](/patterns/passwords/).
+
+You can ask for a security code every time a user signs in or only once per device, depending on the risk level of your service.
+
+## How it works
+
+Send and ask the user for the security code when they: 
+* create an account
+* sign in later
+
+### When creating an account
+
+When the user creates an account, ask for their password and mobile phone number at the same time. Make it clear if you'll only use the mobile number for sending security codes, or if you'll also use it for other purposes.
+
+After saving the user’s password and mobile phone number, verify their mobile phone number by sending them a text message with a 5 digit code in this format:
+
+{% from "govuk/components/inset-text/macro.njk" import govukInsetText %}
+{{ govukInsetText({
+  text: "12345 is your [service name] security code"
+}) }}
+
+Then ask the user to enter this code:
+
+{{ example({group: "patterns", item: "security-code", example: "default", html: true, nunjucks: true, open: false}) }}
+
+Let users enter the code in whatever format is familiar to them. Allow additional spaces, hyphens and dashes.
+
+Give a time limit of 15 minutes for users to enter the code. If this time limit expires when the user correctly enters the code, display a ‘code has expired’ error message and send the user a new code.
```

Comment:
```suggestion
Give a time limit of 15 minutes for users to enter the code before it expires.

If the user correctly enters a recently expired code, display a ‘code has expired’ error message and send the user a new code.

After 2 hours, the expired code should be treated the same as an incorrect code.
```

Reply:
@calvin-lau-sig7 thanks! The last sentence reads a bit weirdly to me ("When the code has expire for more than 2 hours") – could it be something like:

> When more than 2 hours have passed since the code was originally sent, show the ‘Incorrect security code’ message even if the code is correct.

Might also help to resolve ambiguity over if the 2 hours is additional to the 15 mins (so 2:15) or not?

Reply:
Yeah, that sounds good. I've updated both my suggestions.

I can't seem to figure out how to get it shorter (E.g. "After 2 hours,...) whilst avoiding the ambiguity.

Reply:
Another option for this one — I found it was a lot easier to write from the user perspective instead of following code logic:
```suggestion
Give a time limit of 15 minutes for users to enter the code — the code should expire after this time limit.

If the user enters and expired code that was originally sent more than:
- 15 minutes ago, show a ‘code has expired’ error message and send the user a new code
- 2 hours ago, show a ‘incorrect security code’ message
```

Reply:
@calvin-lau-sig7 looks good to me. One small typo: "and expired code". Feel free to commit after fixing that!